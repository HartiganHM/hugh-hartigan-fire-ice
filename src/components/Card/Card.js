import React, { Component } from 'react';
import PropTypes from 'prop-types';
import '../App/App.css'

class Card extends Component {
  toggleClass = () => {
    console.log('click')
  }

  render() {
    const {
      name,
      founded,
      seats,
      titles,
      coatOfArms,
      ancestralWeapons,
      words,
      swornMembers
    } = this.props.houseData;

    const listOfTitles = titles.map((title, index) => (
      <div key={index}>Title: {title}</div>
    ));

    const listOfWeapons = ancestralWeapons.map((weapon, index) => (
      <div key={index}>Weapon: {weapon}</div>
    ));

    const listOfSeats = seats.map((seat, index) => (
      <div key={index}>Seat: {seat}</div>
    ));

    const listOfMembers = swornMembers.map((member, index) => (
      <div key={index}>Sworn Member: {member}</div>
    ));

    return (
      <div className="Card" onClick={() => this.toggleClass}>
        <h2>{name}</h2>
        <div>Founded: {founded || 'N/A'}</div>
        <div>Words {words || 'N/A'}</div>
        {listOfTitles}
        {listOfSeats}
        <div>Coat of Arms: {coatOfArms}</div>
        {listOfWeapons}
        <div className="hidden">{listOfMembers}</div>
      </div>
    );
  }
}

export default Card;

Card.propTypes = {
  houseData: PropTypes.object
};
