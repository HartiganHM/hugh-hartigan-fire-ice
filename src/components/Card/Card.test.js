import Card from './Card';
import React from 'react';
import { shallow } from 'enzyme';
import mockHousesData from '../../data/mockHousesData';
import { read } from 'fs';

describe('Card tests', () => {
  it('Card should match the snapshot', () => {
    const mockHouseData = mockHousesData[0];
    const renderedCard = shallow(<Card houseData={mockHouseData} />);

    expect(renderedCard).toMatchSnapshot();
  });

  it('Card should not display sworn members by default', () => {
    const mockHouseData = mockHousesData[0];
    const renderedCard = shallow(<Card houseData={mockHouseData} />);
    const expected = 0;

    expect(renderedCard.find('.members').length).toEqual(expected);
  });

  it('Card should display a list of members on click', () => {
    const mockHouseData = mockHousesData[0];
    const renderedCard = shallow(<Card houseData={mockHouseData} />);
    const expected = mockHouseData.swornMembers.length;

    renderedCard.simulate('click');
    expect(renderedCard.find('.members').length).toEqual(8);
  });
});
