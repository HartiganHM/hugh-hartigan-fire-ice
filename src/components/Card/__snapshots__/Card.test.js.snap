// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card tests Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    houseData={
        Object {
            "ancestralWeapons": Array [
              "Lady Forlorn",
            ],
            "cadetBranches": Array [],
            "coatOfArms": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
            "currentLord": "https://www.anapioficeandfire.com/api/characters/681",
            "diedOut": "",
            "founded": "",
            "founder": "https://www.anapioficeandfire.com/api/characters/256",
            "heir": "https://www.anapioficeandfire.com/api/characters/677",
            "name": "House Corbray of Heart's Home",
            "overlord": "https://www.anapioficeandfire.com/api/houses/7",
            "region": "",
            "seats": Array [
              "Heart's Home",
            ],
            "swornMembers": Array [
              "https://www.anapioficeandfire.com/api/characters/255",
              "https://www.anapioficeandfire.com/api/characters/256",
              "https://www.anapioficeandfire.com/api/characters/447",
              "https://www.anapioficeandfire.com/api/characters/654",
              "https://www.anapioficeandfire.com/api/characters/677",
              "https://www.anapioficeandfire.com/api/characters/681",
              "https://www.anapioficeandfire.com/api/characters/871",
              "https://www.anapioficeandfire.com/api/characters/1182",
            ],
            "titles": Array [
              "Lord of Heart's Home",
              "King of the Fingers (historical)",
            ],
            "url": "https://www.anapioficeandfire.com/api/houses/81",
            "words": "",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          House Corbray of Heart's Home
</h2>,
        <div>
          Founded: 
          N/A
</div>,
        <div>
          Words 
          N/A
</div>,
        Array [
          <div>
            Title: 
            Lord of Heart's Home
</div>,
          <div>
            Title: 
            King of the Fingers (historical)
</div>,
        ],
        Array [
          <div>
            Seat: 
            Heart's Home
</div>,
        ],
        <div>
          Coat of Arms: 
          Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)
</div>,
        Array [
          <div>
            Weapon: 
            Lady Forlorn
</div>,
        ],
      ],
      "className": "Card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "House Corbray of Heart's Home",
        },
        "ref": null,
        "rendered": "House Corbray of Heart's Home",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Founded: ",
            "N/A",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Founded: ",
          "N/A",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Words ",
            "N/A",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Words ",
          "N/A",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Title: ",
            "Lord of Heart's Home",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Title: ",
          "Lord of Heart's Home",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Title: ",
            "King of the Fingers (historical)",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Title: ",
          "King of the Fingers (historical)",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Seat: ",
            "Heart's Home",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Seat: ",
          "Heart's Home",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Coat of Arms: ",
            "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Coat of Arms: ",
          "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Weapon: ",
            "Lady Forlorn",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Weapon: ",
          "Lady Forlorn",
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            House Corbray of Heart's Home
</h2>,
          <div>
            Founded: 
            N/A
</div>,
          <div>
            Words 
            N/A
</div>,
          Array [
            <div>
              Title: 
              Lord of Heart's Home
</div>,
            <div>
              Title: 
              King of the Fingers (historical)
</div>,
          ],
          Array [
            <div>
              Seat: 
              Heart's Home
</div>,
          ],
          <div>
            Coat of Arms: 
            Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)
</div>,
          Array [
            <div>
              Weapon: 
              Lady Forlorn
</div>,
          ],
        ],
        "className": "Card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "House Corbray of Heart's Home",
          },
          "ref": null,
          "rendered": "House Corbray of Heart's Home",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Founded: ",
              "N/A",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Founded: ",
            "N/A",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Words ",
              "N/A",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Words ",
            "N/A",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Title: ",
              "Lord of Heart's Home",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Title: ",
            "Lord of Heart's Home",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Title: ",
              "King of the Fingers (historical)",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Title: ",
            "King of the Fingers (historical)",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Seat: ",
              "Heart's Home",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Seat: ",
            "Heart's Home",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Coat of Arms: ",
              "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Coat of Arms: ",
            "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Weapon: ",
              "Lady Forlorn",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Weapon: ",
            "Lady Forlorn",
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
